strax/context.py

```diff
@@ -415,12 +416,56 @@ def _set_plugin_config(self, p, run_id, tolerant=True):
                     parent_name = opt.parent_option_name
 
                     mes = (f'Cannot find "{parent_name}" among the options of the parent.'
-                           f' Either you specified by accident {option_name} as child option' 
+                           f' Either you specified by accident {option_name} as child option'
                            f' or you specified the wrong parent_option_name. Have you specified '
                            'the correct parent option name?')
                     assert parent_name in p.config, mes
                     p.config[parent_name] = option_value
 
+    def _config_hash(self):
+        """
+        Dump the current config to a hash as a sanity check for building
+        the _fixed_plugin_cache. If any item changes in the config, so
+        does this hash.
+        """
+        return hashlib.sha1(str(self.config).encode('ascii')).hexdigest()
+        # return strax.deterministic_hash(self.config)
```

Comment:
I tried using the deterministic hash but it does not like immutibledict. Perhaps there is an other solution but my guess is that this serves the purpose quite well.

Reply:
Unfortunately this is not deterministic, you need to sort any dictionaries and sets to get a deterministic string. Also the string representation may depend on the version of the package of non-bultin values. Why not just fix the hashablize function [here](https://github.com/AxFoundation/strax/blob/ba3eb8e9e3e18f7a9b4a2a855c107478c05f5e2b/strax/utils.py#L254)
```python
from collections.abc import Mapping

254        if isinstance(obj, Mapping): # instead of if isinstance(obj, dict)
...
```

Reply:
Thanks Yossi, sure, let me try that. 

Actually, it's not much of an issue if it's non-deterministic, the contexts will always just build the `_config_hash` on the fly, so any changes in order of the config or the version of a package may lead to a different context_hash but as long as it does not change for every run, it shouldn't matter. But you are right, better do make it deterministic because even if there is no issue now because some day it might if one assumes it is deterministic (which shouldn't be a bad assumption).

Reply:
@jmosbacher , I tried your suggestion but it doesn't work. I initially also tried something like this (by doing the the same for `immutabledict` instead of `Mapping`), however, it turns out that since immutabledict has a hash method, this whole logic is never accessed. 

I'll propose something similar to get it working.

Reply:
Oh, sorry about that. I assumed the issue was with the hash function but I guess it was in the json encoder all along? In that case maybe add the Mapping logic to the NumpyJSONEncoder class?